conan_provider.cmake

```diff
@@ -66,15 +66,22 @@ function(detect_arch ARCH)
             message(WARNING "CMake-Conan: Multiple architectures detected, this will only work if Conan recipe(s) produce fat binaries.")
         endif()
     endif()
-    if(CMAKE_SYSTEM_PROCESSOR MATCHES "aarch64|ARM64|arm64" OR CMAKE_OSX_ARCHITECTURES MATCHES arm64)
+    if(CMAKE_SYSTEM_NAME MATCHES "Darwin|iOS|tvOS|watchOS")
+        set(host_arch ${CMAKE_OSX_ARCHITECTURES})
+    elseif(CMAKE_SYSTEM_NAME MATCHES "Windows")
```

Comment:
would it not make sense to check for `MSVC` instead? Based on the discussions in CMake, my understanding was that `CMAKE_CXX_COMPILER_ARCHITECTURE_ID` is well defined for msvc, but may not be for others - assuming other compilers may be used in Windows, like gcc

Reply:
Yes, fixed